commons/all/src/main/java/org/infinispan/commons/util/TypedProperties.java

```diff
@@ -120,19 +133,24 @@ public boolean getBooleanProperty(String key, boolean defaultValue, boolean doSt
    }
 
    public <T extends Enum<T>> T getEnumProperty(String key, Class<T> enumClass, T defaultValue, boolean doStringReplace) {
-      String value = getProperty(key);
-      if (value == null) return defaultValue;
-      value = value.trim();
-      if (value.length() == 0) return defaultValue;
-
-      if (doStringReplace)
-         value = StringPropertyReplacer.replaceProperties(value);
-
-      try {
-         return Enum.valueOf(enumClass, value);
-      } catch (IllegalArgumentException e) {
-         log.unableToConvertStringPropertyToEnum(value, defaultValue.name());
-         return defaultValue;
+      Object value = this.get(key);
+      if (value instanceof Enum) {
+         return (T) value;
```

Comment:
can throw `ClassCastException`. check if the `enum` is the proper `enumClass`
add a proper tests, i.e store "Color" and try to get another enum.

Reply:
Do you prefer a value as null instead of `ClassCastException`?
I do prefer have a `ClassCastException`

Reply:
well, I think the If could be changed to `value != null && value.getClass().isAssignableFrom(enumClass)` or similar. No exception handling needed.